pkg/controllers/dynakube/logmodule/daemonset/container_test.go

```diff
@@ -0,0 +1,126 @@
+package daemonset
+
+import (
+	"testing"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetContainer(t *testing.T) {
+	t.Run("get main container", func(t *testing.T) {
+		dk := dynakube.DynaKube{}
+		mainContainer := getContainer(dk)
+
+		require.NotEmpty(t, mainContainer)
+
+		assert.NotEmpty(t, mainContainer.Name)
+		assert.NotEmpty(t, mainContainer.Image)
+		assert.NotEmpty(t, mainContainer.ImagePullPolicy)
+		assert.NotEmpty(t, mainContainer.VolumeMounts)
+		assert.Len(t, mainContainer.VolumeMounts, expectedMountLen)
+		assert.NotEmpty(t, mainContainer.Env)
+		assert.Len(t, mainContainer.Env, expectedBaseEnvLen)
+		assert.NotEmpty(t, mainContainer.SecurityContext)
+	})
+
+	t.Run("image-ref is respected", func(t *testing.T) {
+		expectedRepo := "my-test-repo"
+		expectedTag := "my-test-tag"
+		dk := dynakube.DynaKube{}
+		dk.Spec.Templates.LogModule.ImageRef = dynakube.ImageRefSpec{
+			Repository: expectedRepo,
+			Tag:        expectedTag,
+		}
+		mainContainer := getContainer(dk)
+
+		require.NotEmpty(t, mainContainer)
+		assert.NotEmpty(t, mainContainer.Image)
+		assert.Equal(t, expectedRepo+":"+expectedTag, mainContainer.Image)
+	})
+}
+
+func TestGetInitContainer(t *testing.T) {
+	t.Run("get main container", func(t *testing.T) {
+		dk := dynakube.DynaKube{}
+		initContainer := getInitContainer(dk)
+
+		require.NotEmpty(t, initContainer)
+
+		assert.NotEmpty(t, initContainer.Name)
+		assert.NotEmpty(t, initContainer.Image)
+		assert.NotEmpty(t, initContainer.ImagePullPolicy)
+		assert.NotEmpty(t, initContainer.Args)
+		assert.Len(t, initContainer.Args, expectedBaseInitArgsLen)
+		assert.NotEmpty(t, initContainer.Command)
+		assert.NotEmpty(t, initContainer.VolumeMounts)
+		assert.Len(t, initContainer.VolumeMounts, expectedInitMountLen)
+		assert.NotEmpty(t, initContainer.Env)
+		assert.Len(t, initContainer.Env, expectedBaseInitEnvLen)
+		assert.NotEmpty(t, initContainer.SecurityContext)
+	})
+
+	t.Run("image-ref is respected", func(t *testing.T) {
+		expectedRepo := "my-test-repo"
+		expectedTag := "my-test-tag"
+		dk := dynakube.DynaKube{}
+		dk.Spec.Templates.LogModule.ImageRef = dynakube.ImageRefSpec{
+			Repository: expectedRepo,
+			Tag:        expectedTag,
+		}
+		initContainer := getContainer(dk)
+
+		require.NotEmpty(t, initContainer)
+		assert.NotEmpty(t, initContainer.Image)
+		assert.Equal(t, expectedRepo+":"+expectedTag, initContainer.Image)
+	})
+}
+
+func TestSecurityContext(t *testing.T) {
+	t.Run("get base securityContext", func(t *testing.T) {
+		dk := dynakube.DynaKube{}
+		sc := getBaseSecurityContext(dk)
+
+		require.NotNil(t, sc)
+		require.NotEmpty(t, sc)
+
+		assert.NotNil(t, sc.Privileged)
```

Comment:
Could we verify specific values True/False? It's hard requirement that Privileged is False. The same regarding other SecurityContext properties. We will be warned if want to change the default setting.

Reply:
IMO, that doesn't help more

I agree these tests are "tests for test's sake".
What these tests check is: "do we explicitly set the fields in the securityContext?"
- this is the "most sensible" test I could come up with
- I would be more OK with removing these tests(to reduce noise), then adding more exact ones, as that feels like having a test for checking the value of a `const`, which IMO is totally unnecessary 😅 

"dynamic" fields have a separate tests, to see if they are actually dynamic. (these are actual tests)